Compiling for android platform

Hi! I want to use your library for another rust project which compiled for android platform. Unfortunately when I try to build my project, I have a following error:

`cargo:warning=simd_c/reedsolomon.c:1:0: error: unknown value 'native' for -march`

I inspected your code and found a problem: gcc compler for android (aarch64-linux-android-gcc) doesn't support -march flag. Project builds successfully when I remove this flag from the build.rs file. Could you please add some option for your library to disable this compiler flag?

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -9,6 +9,8 @@
     - The `2.0.X` and `1.X.X` data are simply adjusted by mutiplying `10^6` then dividing by `2^20`
 - Dependencies update
   - Updated `rand` from `0.4` to `0.5.4`
+- Added special handling in `build.rs` for CC options on Android
+  - `-march=native` is not available for GCC on Android, see issue #23
 
 ## 3.1.0
 - Impl'd `std::error::Error` for `reed_solomon_erasure::Error` and `reed_solomon_erasure::SBSError`

--- a/build.rs
+++ b/build.rs
@@ -157,12 +157,20 @@ fn write_tables() {
     all(not(feature = "pure-rust"),
         any(target_arch = "x86_64", target_arch = "aarch64")))]
 fn compile_simd_c() {
-    cc::Build::new()
-        .opt_level(3)
-        .flag("-march=native")
-        .flag("-std=c11")
-        .file("simd_c/reedsolomon.c")
-        .compile("reedsolomon");
+    if cfg!(target_os = "android") {
+        cc::Build::new()
+            .opt_level(3)
+            .flag("-std=c11")
+            .file("simd_c/reedsolomon.c")
+            .compile("reedsolomon");
+    } else {
+        cc::Build::new()
+            .opt_level(3)
+            .flag("-march=native")
+            .flag("-std=c11")
+            .file("simd_c/reedsolomon.c")
+            .compile("reedsolomon");
+    }
 }
 
 #[cfg(

--- a/build.rs
+++ b/build.rs
@@ -157,7 +157,9 @@ fn write_tables() {
     all(not(feature = "pure-rust"),
         any(target_arch = "x86_64", target_arch = "aarch64")))]
 fn compile_simd_c() {
-    if cfg!(target_os = "android") {
+    if     cfg!(target_os = "android")
+        || cfg!(target_os = "ios")
+    {
         cc::Build::new()
             .opt_level(3)
             .flag("-std=c11")

--- a/src/lib_tests.rs
+++ b/src/lib_tests.rs
@@ -5,14 +5,6 @@ use super::*;
 use shard_utils;
 use rand::{thread_rng, Rng};
 
-macro_rules! shards {
-    (
-        $( [ $( $x:expr ),* ] ),*
-    ) => {{
-        vec![ $( vec![ $( $x as u8),* ].into_boxed_slice() ),* ]
-    }}
-}
-
 macro_rules! make_random_shards {
     ($per_shard:expr, $size:expr) => {{
         let mut shards = Vec::with_capacity(20);

--- a/src/lib_tests.rs
+++ b/src/lib_tests.rs
@@ -5,7 +5,7 @@ use super::*;
 use shard_utils;
 use rand::{thread_rng, Rng};
 
-macro_rules! shards {
+macro_rules! shards_for_test {
     (
         $( [ $( $x:expr ),* ] ),*
     ) => {{
@@ -857,16 +857,16 @@ fn test_reconstruct_error_handling() {
 fn test_one_encode() {
     let r = ReedSolomon::new(5, 5).unwrap();
 
-    let mut shards = shards!([0, 1],
-                             [4, 5],
-                             [2, 3],
-                             [6, 7],
-                             [8, 9],
-                             [0, 0],
-                             [0, 0],
-                             [0, 0],
-                             [0, 0],
-                             [0, 0]);
+    let mut shards = shards_for_test!([0, 1],
+                                      [4, 5],
+                                      [2, 3],
+                                      [6, 7],
+                                      [8, 9],
+                                      [0, 0],
+                                      [0, 0],
+                                      [0, 0],
+                                      [0, 0],
+                                      [0, 0]);
 
     r.encode_shards(&mut shards).unwrap();
     { assert_eq!(shards[5][0], 12);
@@ -1086,10 +1086,10 @@ fn test_check_slices_or_shards_size() {
     let r = ReedSolomon::new(2, 2).unwrap();
 
     {
-        let mut shards = shards!([0, 0, 0],
-                                 [0, 1],
-                                 [1, 2, 3],
-                                 [0, 0, 0]);
+        let mut shards = shards_for_test!([0, 0, 0],
+                                          [0, 1],
+                                          [1, 2, 3],
+                                          [0, 0, 0]);
 
         assert_eq!(Error::IncorrectShardSize,
                    r.encode_shards(&mut shards)
@@ -1105,10 +1105,10 @@ fn test_check_slices_or_shards_size() {
                    .unwrap_err());
     }
     {
-        let mut shards = shards!([0, 1],
-                                 [0, 1],
-                                 [1, 2, 3],
-                                 [0, 0, 0]);
+        let mut shards = shards_for_test!([0, 1],
+                                          [0, 1],
+                                          [1, 2, 3],
+                                          [0, 0, 0]);
 
         assert_eq!(Error::IncorrectShardSize,
                    r.encode_shards(&mut shards)
@@ -1124,10 +1124,10 @@ fn test_check_slices_or_shards_size() {
                    .unwrap_err());
     }
     {
-        let mut shards = shards!([0, 1],
-                                 [0, 1, 4],
-                                 [1, 2, 3],
-                                 [0, 0, 0]);
+        let mut shards = shards_for_test!([0, 1],
+                                          [0, 1, 4],
+                                          [1, 2, 3],
+                                          [0, 0, 0]);
 
         assert_eq!(Error::IncorrectShardSize,
                    r.encode_shards(&mut shards)
@@ -1143,10 +1143,10 @@ fn test_check_slices_or_shards_size() {
                    .unwrap_err());
     }
     {
-        let mut shards = shards!([],
-                                 [0, 1, 3],
-                                 [1, 2, 3],
-                                 [0, 0, 0]);
+        let mut shards = shards_for_test!([],
+                                          [0, 1, 3],
+                                          [1, 2, 3],
+                                          [0, 0, 0]);
 
         assert_eq!(Error::EmptyShard,
                    r.encode_shards(&mut shards)

--- a/src/lib_tests.rs
+++ b/src/lib_tests.rs
@@ -5,6 +5,14 @@ use super::*;
 use shard_utils;
 use rand::{thread_rng, Rng};
 
+macro_rules! shards {
+    (
+        $( [ $( $x:expr ),* ] ),*
+    ) => {{
+        vec![ $( vec![ $( $x as u8),* ].into_boxed_slice() ),* ]
+    }}
+}
+
 macro_rules! make_random_shards {
     ($per_shard:expr, $size:expr) => {{
         let mut shards = Vec::with_capacity(20);

